lib/scrabble/scrabble.rb

```diff
@@ -0,0 +1,81 @@
+module Scrabble
+  class Scrabble
+
+    ONE_POINT = ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"]
+    TWO_POINTS = ["D", "G"]
+    THREE_POINTS = ["B", "C", "M", "P"]
+    FOUR_POINTS = ["F", "H", "V", "W", "Y"]
+    FIVE_POINTS = ["K"]
+    EIGHT_POINTS = ["J", "X"]
+    TEN_POINTS = ["Q", "Z"]
+
+    # RETURNS THE SCORE OF THE WORD
+    def self.score(word)
+      return 0 if valid?(word) == false
+      word.upcase!  # done so I don't need to .upcase every time
+
+      word_array = word.split(//)  # breaks word into an array of letters
+      points = 0
+
+      # Adds respective value to score if letter is included in the constant.
+      word_array.each do |letter|
+        points += 1 if ONE_POINT.include?(letter)
+        points += 2 if TWO_POINTS.include?(letter)
+        points += 3 if THREE_POINTS.include?(letter)
+        points += 4 if FOUR_POINTS.include?(letter)
+        points += 5 if FIVE_POINTS.include?(letter)
+        points += 8 if EIGHT_POINTS.include?(letter)
+        points += 10 if TEN_POINTS.include?(letter)
+      end
+
+      return points
+    end
+
+    # RETURNS THE WORD IN THE ARRAY WITH THE HIGHEST SCORE
+    def self.highest_score_from(array_of_words)
+      return 0 if valid_array?(array_of_words)
+
+      array_of_words.each { |word| word.upcase! }  # upcase every word
+      highest_scoring_word = array_of_words[0]
+
+      array_of_words.each do |word|
+        if score(highest_scoring_word) < score(word)      # checks if the score of the highest scoring word is less than the word
+          highest_scoring_word = word                     # if so, make the word the new highest scoring word
+        elsif score(highest_scoring_word) == score(word)  # if the scores of the words are equal
+          if highest_scoring_word.length < 7 && word.length == 7
+            highest_scoring_word = word
+          elsif highest_scoring_word.length == 7 && word.length < 7
+            highest_scoring_word = highest_scoring_word # redunant I know, easier for me to undestand the logic
```

Comment:
re: that comment - that's a totally valid reason to make a method a little longer! The other option is a comment explaining the logic, but I prefer clarity in the code.